packages/cli/src/lib/push/caseInsensitiveDocument.ts

```diff
@@ -0,0 +1,8 @@
+export class CaseInsensitiveDocument<T> {
```

Comment:
I'm using case insensitive check on "required keys" and "known keys", in order for  `documentid, documentId, DocumentID` etc to all be equal and "correct".

Less annoying for users IMO.